src/pages/NotificationsMenu.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { makeStyles } from "@material-ui/core";
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import NotificationsIcon from '@material-ui/icons/Notifications';
+import ListItemText from "@material-ui/core/ListItemText";
+import TableCell from "@material-ui/core/TableCell";
+import {createMuiTheme, ThemeProvider} from "@material-ui/core";
+import NotificationStyles from "../NotificationStyles";
+const useStyles = makeStyles(NotificationStyles);
+const setDate = '2020-08-09 02:07:10';              // example date
+
+/*
+   This function takes a string in format `yyyy-mm-dd  hh:mm:ss` and returns a date object with that
+   time
+ */
+function dateStringToObj(dateStr) {
+
+    const dateObj = new Date();
+    if (typeof dateStr === 'string') {
+        const parts = dateStr.split(" "),
+            datePart = parts[0].split("-"),
+            timePart = parts[1].split(":");
+        const dateObj = new Date(datePart[0], datePart[1] - 1, datePart[2], timePart[0], timePart[1], timePart[2]);
+        return dateObj
+    }
+    return dateObj;
+}
+
+/*
+ This function takes in a string representation of a date object and returns the
+ time difference in the simplest form
+ */
+function dateDifference(dateObj) {
+
+    // const randDate = '2020-03-01 02:07:10';              // example date
+    const diff = (new Date().getTime() -  dateObj.getTime()) / 1000,
+        dayDiff = Math.floor(diff / 86400);
+
+    // if dayDiff is invalid or less than 0
+    if (isNaN(dayDiff) || (dayDiff < 0))
+        return;
+
+    return dayDiff == 0 && (
+        diff < 60 && "Just Now" ||
+        diff < 120 && "A minute ago" ||
+        diff < 3600 && Math.floor( diff / 60 ) + " minutes ago" ||
+        diff < 7200 && "An hour ago" ||
+        diff < 86400 && Math.floor( diff / 3600 ) + " hours ago") ||
+        dayDiff == 1 && "Yesterday" ||
+        dayDiff < 7 && dayDiff + " days ago" ||
+        dayDiff > 31 && Math.ceil(dayDiff / 7) + " weeks ago";
```

Comment:
As far as I could find date-fns does not have one single function to help with making the date simple but it has functions to subtract days and time. I'll be replacing the calculations with those functions but the main "framework" of the function will remain the same.

Reply:
so, date-fns definitely does have this! In fact, it has several variations depending on needs: [`formatDistance`](https://date-fns.org/v2.15.0/docs/formatDistance), [`formatDistanceStrict`](https://date-fns.org/v2.15.0/docs/formatDistanceStrict), and [`formatRelative`](https://date-fns.org/v2.15.0/docs/formatRelative), plus the two `Distance` ones also have `DistanceToNow` variants that automatically pull in "now" so you don't have to pass that in manually -- these functions appear in two different places I could find -- in the `i18n` section, but also in the "common helpers" below that in the documentation sidebar. One thing to note which may or may not be relevant to you, is that elsewhere we're using date-fns' `lightFormat` rather than `format` -- this was a size tradeoff; lightFormat doesn't support every token and I think doesn't support locales. The `Distance` and `Relative` functions don't appear to have equivalents that differ in the same ways, which may mean that they'll pull in the non-`light` version of the regular formatter too, and if they do we may want to switch the other date formatting to use the full-fledged `format` too. That would probably fall outside of what you'd be expected to do, but it's the sort of thing I like to mention at times like this.

Reply:
I totally skipped that. Thank you for informing. I was searching for something along the lines of "Subtract time" and clearly did not browse through the docs well enough. I did not completely understand what you meant by 
> "The Distance and Relative functions don't appear to have equivalents that differ in the same ways, which may mean that they'll pull in the non-light version of the regular formatter too". 
What problems would we face exactly?

Reply:
The `lightFormat` function is different in only one way that matters to us, basically, which is that it's smaller as far as transfer size. If we have need of another function that happens to bring in the full version of `format` (which, it's worth noting, I don't know for sure that these do! I'm just guessing they might), then there's no benefit of us using `lightFormat` any more, and in fact might be slightly to our detriment since we'd be using both versions.

Basically, it's a small performance benefit to not send as much javascript to our users, so it's better, one way or another, to only be using one of the two. But it won't cause any immediate bugs.